### ISSUE TITLE
Opacity not working as expected with MediaElementPlayer and WebView2

### ISSUE THREAD

**Issue by gemelo-Peer** [microsoft/microsoft-ui-xaml#8130]
### Describe the bug

Unlike expected, the Opacity property does not change the transparency of MediaPlayerElement and WebView2 controls, but only changes between visible (opacity > 0.0) and invisible (opacity == 0.0). This also applies if the opacity is set on a parent or grand parent control.

However, the other way round, it is possible to place partially transparent elements over one of the controls mentioned. This is not the well-known airspace problem that sometimes occurs with WPF in conjunction with certain controls. Nevertheless, I fear that the problem is by-design.

Nevertheless, I would like to have an explanation for the problem and would like to ask if there are plans to change this behavior. Regardless, I think it should be documented so that one is not surprised by the problem.

In my case, it is about cross-fading two videos. Maybe someone has an idea how I can do this after all :)

### Steps to reproduce the bug

1. Add a MediaPlayerElement to your UserControl
2. Set Opacity to 0.5.
3. The MediaPlayerElement would be fully opaque.

### Expected behavior

The expected behavior would be that the MediaPlayerElement gets semi-transparent.

### NuGet package version

None

### Additional context

Microsoft.WindowsAppSDK 1.2.230118.102

**Comment by DominicMaas** [comment]
I'm trying to clip a media player element with a corner radius which is not working as expected. (I think a similar issue to here, at least with the design of `MediaPlayerElement`)

I recall reading / hearing from somewhere that `MediaPlayerElement` is technically punching through the WinUI window and is not rendered natively? I may be mis-understanding though.

**Comment by codendone** [comment]
> I recall reading / hearing from somewhere that MediaPlayerElement is technically punching through the WinUI window and is not rendered natively?

That's correct: "External" content like MediaPlayerElement (which renders via a DirectX swapchain), WebView2, and SwapChainPanel (also a swapchain) are handled by punching a hole (erasing any pixels in the WinUI3/WinAppSDK rendering to be transparent in the area of these elements) and hosting the MPE/WV2/SCP below the surface which contains all the WinUi3/WinAppSDK rendering. This prevents opacity from working on these elements.

For clipping/corner-radius, I think clipping might work depending on where the clipping is applied.

**Comment by DominicMaas** [comment]
Makes sense. Thanks!

In terms of clipping, I've opened issue #8264 as it does work (ish), but not consistently. Not really related to this issue.

**Comment by gemelo-Peer** [comment]
Thank you for your clarification. However, in my opinion, there is still a lack of sufficient documentation for this phenomenon.

**Comment by PureAVLtd** [comment]
Although the explanation makes sense. The UWP did not suffer this issue and we often transitioned / faded video in / out as part of our presentation applications with no issue.

**Comment by softworkz** [comment]
> That's correct: "External" content like MediaPlayerElement (which renders via a DirectX swapchain), WebView2, and SwapChainPanel (also a swapchain) are handled by punching a hole (erasing any pixels in the WinUI3/WinAppSDK rendering to be transparent in the area of these elements) and hosting the MPE/WV2/SCP below the surface which contains all the WinUi3/WinAppSDK rendering. This prevents opacity from working on these elements.

@codendone - I have to agree that this is a huge disappointment as it doesn't keep the promises of the new app-local composition model. 
We have an application which relies on having a transparent webview on top of a media player element. We have this working on a wide range of platforms:

- Electron (Windows, Linux, Max)
- Android
- iOS
- Tizen (Samsung)
- WebOs (LG)
- UWP+WinUI2 (Windows/Xbox)
- UWP+WinJS (previously for Windows+Xbox)

WinUI3 is the only UI framework where this isn't possible.

**How could it happen that MS made such a big conceptual mistake in the design of WinUI3 composition?**
